### DISCUSSION TITLE
CTkToolTips

### DISCUSSION THREAD

**Discussion by Akascape** [TomSchimansky/CustomTkinter#1425] [Showcase]
Get this Tooltips for your application: https://github.com/Akascape/CTkToolTip

![example](https://user-images.githubusercontent.com/89206401/229771216-199727ef-2a01-4ab3-bac6-7e0a5234389e.gif)

## Features
- Display over any widget
- Configurable options
- Transparency effect
- Round corners
- Can show live value of a CTk Slider
- Add delays

**Comment by RunsOnRum** [comment]
This is brilliant and hope it gets integrated into CTk. I do have one issue though, which is an issue that I have come across in other Tool Tips for tkinter. If the object gets destroyed, the tooltip will still be displayed after the delay but the object that it belongs to is gone and the tooltip will remain active. This is alleviated by setting delay to 0 but not always wanted.

In my scenario, I have a toplevel window that closes on a button being pressed. If I have the tooltip with a delay set, the windows closes, then the popup appears.

I hope that makes sense. Thank you for your great work and your thoughtful information in these discussions.  

Sorry.. two issues:
If you have a border set to 1 with pady set to 0, you get no top border. Also, in the same regard padx set to 0 will cause the side borders to disappear
![image](https://user-images.githubusercontent.com/9207350/231074513-43cccda0-443f-48a1-af91-676ad703821a.png)

**Comment by Akascape** [comment]
@RunsOnRum 

Please download the new version, I have just fixed these issues.

**Reply by RunsOnRum** [reply]
Legend! Thank you

**Comment by avalon60** [comment]
This looks really nice!

**Reply by avalon60** [reply]
Switched over to this for my CustomTkinter widgets - superb!

**Comment by RunsOnRum** [comment]
Hello @Akascape, I'm getting a performance hit when attaching tooltips to a generated list of buttons. I haven't put much thought toward what I can do to get around it yet (it probably is just due to my abuse of the tooltips :-)). I figured I would mention in case you could pinpoint would it could be quickly.

**Reply by Akascape** [reply]
@RunsOnRum That was because of CTkToplevel, I have replaced it with the normal tk toplevel and it works the same and also fixes the performance issue. Please download the new version...

**Reply by RunsOnRum** [reply]
Wow!! Thank you for such a quick response. Very much appreciated. Works like a charm!

**Reply by Akascape** [reply]
@RunsOnRum Please update again, I just fixed another bug 😅

**Reply by RunsOnRum** [reply]
Thank you so much, I will implement in the morning.

**Reply by RunsOnRum** [reply]
@Akascape, This update is putting all the tooltips in the top left corner of my screen on application startup. They stay there until I hover over the buttons. The tooltips also still appear in the location I desire when hovering over button. I find it strange that pycharm run does not produce the same results and compiling with pyinstaller, as the tooltips function correctly when run with pycharm.

**Reply by Akascape** [reply]
@RunsOnRum It was again a bug but I fixed it.
